src/core/api/storage.rs

```diff
@@ -20,46 +20,46 @@ use crate::{
 	core::storage::{RecordsStorage, RecordsStorageConfig, StorageEntry},
 	eyre,
 };
-use subxt::sp_core::H256;
+use std::{fmt::Debug, hash::Hash};
 use tokio::sync::{
 	mpsc::{Receiver, Sender},
 	oneshot,
 };
 
 // Storage requests
 #[derive(Clone, Debug)]
-pub enum RequestType {
-	Write(H256, StorageEntry),
-	Read(H256),
-	Replace(H256, StorageEntry),
+pub enum RequestType<K: Ord + Hash + Sized> {
+	Write(K, StorageEntry),
+	Read(K),
+	Replace(K, StorageEntry),
 	Size,
-	Keys,
+	Keys(Option<K>),
```

Comment:
It would be more explicit to have a variant `KeysWithPrefix` which also gets rid of the `Option`

Reply:
Yes, I was thinking about it, good point.